Framework/src/AggregatorRunner.cxx

```diff
@@ -0,0 +1,227 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   AggregatorRunner.cxx
+/// \author Barthelemy von Haller
+///
+
+#include "QualityControl/AggregatorRunner.h"
+
+#include <fairlogger/Logger.h>
+// O2
+#include <Common/Exceptions.h>
+#include <Configuration/ConfigurationFactory.h>
+#include <Framework/DataSpecUtils.h>
+#include <Monitoring/MonitoringFactory.h>
+#include <Monitoring/Monitoring.h>
+#include <Framework/InputRecordWalker.h>
+// QC
+#include "QualityControl/DatabaseFactory.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/ServiceDiscovery.h"
+#include "QualityControl/Aggregator.h"
+
+using namespace AliceO2::Common;
+using namespace AliceO2::InfoLogger;
+using namespace o2::framework;
+using namespace o2::configuration;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::repository;
+using namespace std;
+using namespace o2::monitoring;
+
+const auto current_diagnostic = boost::current_exception_diagnostic_information;
+
+namespace o2::quality_control::checker
+{
+
+AggregatorRunner::AggregatorRunner(const std::string& configurationSource, const vector<framework::OutputSpec> checkRunnerOutputs)
+  : mDeviceName(createAggregatorRunnerName()),
+    mTotalNumberObjectsReceived(0)
+{
+  try {
+    mConfigFile = ConfigurationFactory::getConfiguration(configurationSource);
+  } catch (...) {
+    // catch the exceptions and print it (the ultimate caller might not know how to display it)
+    ILOG(Fatal) << "Unexpected exception during initialization:\n"
+                << boost::current_exception_diagnostic_information(true) << ENDM;
+    throw;
+  }
+
+  // prepare list of all inputs
+  int i = 0;
+  for (const auto& spec : checkRunnerOutputs) {
+    auto input = DataSpecUtils::matchingInput(spec);
+    input.binding = "checkerOutput" + to_string(i);
```

Comment:
why not use the `binding` from `spec` to make it simpler and clearer?

Reply:
Because it is empty. I had talked to Giulio but I don't remember the exact reason. I have added a comment to explain it.

Reply:
ok